tasks/setup-Debian.yml

```diff
@@ -27,25 +27,24 @@
     state: present
   when: ansible_distribution == 'Ubuntu' and ansible_distribution_version is version('20.04', '>=')
 
-- name: Add Docker apt key.
-  ansible.builtin.get_url:
-    url: "{{ docker_apt_gpg_key }}"
-    dest: /etc/apt/trusted.gpg.d/docker.asc
-    mode: '0644'
-    force: false
-    checksum: "{{ docker_apt_gpg_key_checksum | default(omit) }}"
-  register: add_repository_key
-  ignore_errors: "{{ docker_apt_ignore_key_error }}"
-  when: docker_add_repo | bool
+- name: Ensure directory exists for /etc/apt/keyrings
+  file:
+    path: /etc/apt/keyrings
+    state: directory
+    mode: '0755'
 
-- name: Ensure curl is present (on older systems without SNI).
+- name: Ensure curl is present
   package: name=curl state=present
-  when: add_repository_key is failed and docker_add_repo | bool
 
-- name: Add Docker apt key (alternative for older systems without SNI).
+- name: Add Docker apt key
   shell: >
-    curl -sSL {{ docker_apt_gpg_key }} | apt-key add -
-  when: add_repository_key is failed and docker_add_repo | bool
+    curl -fsSL {{ docker_apt_gpg_key }} | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg --yes
+  changed_when: false
```

Comment:
not sure of a way around this

Reply:
Yes, there is: https://stackoverflow.com/questions/71585303/how-can-i-manage-keyring-files-in-trusted-gpg-d-with-ansible-playbook-since-apt

You can just skip the `gpg --dearmor` and save the file directly into `/etc/apt/keyrings/` with a `.asc` extension. This means you can skip the `shell` module completely and only use `get_url`.

Reply:
Really all that needs to be changed here is the `dest: /etc/apt/trusted.gpg.d/docker.asc` from the `Add Docker apt key.` task needs to be `dest: /etc/apt/keyrings/docker.asc` instead.